Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs

```diff
@@ -44,23 +44,36 @@ public enum RecognizerStartBehavior { AutoStart, ManualStart };
         public KeywordAndResponse[] KeywordsAndResponses;
 
         private KeywordRecognizer keywordRecognizer;
-        private Dictionary<string, UnityEvent> responses;
+        private readonly Dictionary<string, UnityEvent> responses = new Dictionary<string, UnityEvent>();
 
         void Start()
         {
-            if (KeywordsAndResponses.Length > 0)
+            int keywordCount = KeywordsAndResponses.Length;
+            if (keywordCount > 0)
             {
-                // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
-                // This helps easily link the keyword recognized to the UnityEvent to be invoked.
-                responses = KeywordsAndResponses.ToDictionary(keywordAndResponse => keywordAndResponse.Keyword,
-                                                              keywordAndResponse => keywordAndResponse.Response);
-
-                keywordRecognizer = new KeywordRecognizer(responses.Keys.ToArray());
-                keywordRecognizer.OnPhraseRecognized += KeywordRecognizer_OnPhraseRecognized;
-
-                if (RecognizerStart == RecognizerStartBehavior.AutoStart)
+                try
                 {
-                    keywordRecognizer.Start();
+                    string[] keywords = new string[keywordCount];
+                    // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
+                    // This helps easily link the keyword recognized to the UnityEvent to be invoked.
+                    for (int index = keywordCount; --index >= 0;)
```

Comment:
I run a [benchmark](https://gist.github.com/aalmada/b535b09ed83092b9288ff55d59b3b656) and to iterate an array of 100000000 ints, it takes:
foreach: 972ms
for: 742ms
reversed for: 566ms
It confirms my experience but, in this case, the number of keywords is MUCH smaller so readability may prevail. Should I change it? To a foreach or a for?